packages/webapp/src/containers/Finances/TransactionFilter/index.jsx

```diff
@@ -38,6 +38,26 @@ const TransactionFilter = () => {
   const transactionsFilter = useSelector(transactionsFilterSelector);
   const dispatch = useDispatch();
 
+  const handleChangeFilters = (filterKey, filterState) => {
+    // Check if filterState (filterKey "EXPENSE_TYPE" or "REVENUE_TYPE") has active filters:
+    const hasActiveFilters = Object.values(filterState).some((filter) => {
+      return filter.active === true;
+    });
+
+    // If filterState has no active filters, reset the value of its filterKey on 'tempTransactionsFilter' to undefined:
+    if (!hasActiveFilters) {
+      setTempTransactionsFilter((prevState) => {
+        prevState[filterKey] = undefined;
+        return prevState;
+      });
+    } else {
+      setTempTransactionsFilter((prevState) => {
+        prevState[filterKey] = filterState;
+        return prevState;
+      });
+    }
+  };
```

Comment:
`prevState[filterKey] = ...` mutates the state directly, which should be avoided.
https://react.dev/reference/react/useState#my-initializer-or-updater-function-runs-twice

Also, you can combine these two by moving the conditions inside `setTempTransactionsFilter` or using a ternary operator. (I prefer the latter!)